### ISSUE TITLE
[Feature Request] Data API for streaming

### ISSUE THREAD

**Issue by PachiSystems** [TeamRizu/OutFox#660]
Not sure if that's the correct title for this, but the idea is as follows:

for those that stream on Twitch/Youtube/etc..., it would be nice to be able to have a way to connect to Outfox from another system to grab information like current song/mix, play history, step judgements, total steps in session etc... Can't think of everything that might be fun to put on a stream. Maybe even a way to grab a list of the songs on the streamer's system so they could create some kind of catalog viewers could request from?

Basically the setup that I'm thinking of is Outfox is installed on a dedicated machine and connected to the network whereas the computer hosting the stream is seperate, but also connected to the network. I'm not all that great at putting ideas into words. >_<

**Comment by deltaryz** [comment]
This would be excellent for things like custom light animations and such as well, if it exposes information like the current beat and bpm.

My roommate is currently using a build of StepMania 5.1 on a DDR cab that has the source code modified to talk to an Arduino, which drives a set of LEDs on the cabinet which would
- animate to the beat of the music
- indicate the current health meter of player 1 and 2 individually
- detect the current game mode to change color/animation style

With a game state API, this same behavior could be implemented outside the engine without needing source code access.
